File: BaekJoon Online Judge/C++/Step14-Math_3 Number Theory and Combinatorics/3036.cpp
//
//  3036.cpp
//  Step14-Math_3 Number Theory and Combinatorics
//
//  Created by 김예빈 on 2019. 10. 1..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int gcd(int, int);

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, greater;
    
    cin >> n;
    
    int ring[n];
    
    for(int i = 0; i < n; i++)
        cin >> ring[i];
    
    for(int i = 1; i < n; i++) {
        if(ring[i] > ring[0])
            greater = gcd(ring[i], ring[0]);
        else
            greater = gcd(ring[0], ring[i]);
        
        cout << ring[0] / greater << "/" << ring[i] / greater << "\n";
    }
    
    return 0;
}

int gcd(int a, int b) {
    if(a % b == 0)
        return b;
    else
        return gcd(b, a % b);
}

File: BaekJoon Online Judge/Swift/etc/3036.swift
//
//  3036.swift
//  Step16-Math_3 Number Theory and Combinatorics
//
//  Created by Yebin Kim on 2019/12/30.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

func euclidean(_ a: Int, _ b: Int) -> Int {
    if b % a == 0 {
        return a
    } else {
        return euclidean(b % a, a)
    }
}

let _ = Int(readLine()!)!
let circle = readLine()!.split(separator: " ").map{ Int($0)! }
let first = circle.first!

for i in 1..<circle.count {
    let gcd = euclidean(max(circle[i], first), min(circle[i], first))
    print("\(first / gcd)/\(circle[i] / gcd)")
}
